app/src/main/java/com/asfoundation/wallet/promotions/PromotionsPresenter.kt

```diff
@@ -54,7 +54,11 @@ class PromotionsPresenter(private val view: PromotionsView,
       }
       promotionsModel.walletOrigin == WalletOrigin.UNKNOWN -> {
         viewState = ViewState.UNKNOWN
-        view.showLockedPromotionsScreen()
+        if (promotionsModel.vouchers.isNotEmpty()) {
+          view.showLockedPromotionsWithVouchers(promotionsModel.vouchers)
+        } else {
+          view.showLockedPromotionsScreen()
+        }
```

Comment:
I don't agree because it should not depende on if it has vouchers or not, but rather of the wallet origin. But maybe a name change would be more fit, maybe to UNKNOWN_USER.
The purpose of the viewState as of right now it's to update the view onResume, to handle the possibility of the user sending the app to background, make a purchase and then go back to the app and be able to refresh the view automatically. And for that I believe that the differente between the two would be redundant.